Data/Vector/Generic/Base.hs

```diff
@@ -1,5 +1,5 @@
 {-# LANGUAGE Rank2Types, MultiParamTypeClasses, FlexibleContexts,
-             TypeFamilies, ScopedTypeVariables, BangPatterns #-}
+             TypeFamilies, ScopedTypeVariables, BangPatterns, DeriveFunctor #-}
```

Comment:
I couldn't find where `DeriveFunctor` is being used in this PR? Also, according to ghc manual `DeriveFunctor` is available since ghc-7.10, if that extension is necessary for something are we dropping support for older ghc? If, yes, I am in favor, but we'll have to coordinate all the cleanup in a separate ticket.

Reply:
That's remnant of `Box` monad, then I found that vector already has one and DeriveFunctor remained. I'll remove it.

Reply:
(We've already dropped GHC < 7.10)

Reply:
@Bodigrim That's right, I remember now reviewing the PR #317 
Thanks for reminding me ;)